src/sentry/grouping/enhancer/__init__.py

```diff
@@ -330,7 +335,13 @@ def as_dict(self):
             matchers[matcher.key] = matcher.pattern
         return {"match": matchers, "actions": [str(x) for x in self.actions]}
 
-    def get_matching_frame_actions(self, frames, platform, exception_data=None, cache=None):
+    def get_matching_frame_actions(
+        self,
+        match_frames: Sequence[dict[str, Any]],
+        platform: str,
+        exception_data: dict[str, Any],
+        in_memory_cache: dict[str, str],
+    ) -> list[tuple[int, Action]]:
```

Comment:
The return type is based on the following code.
https://github.com/getsentry/sentry/blob/7d67a85faca4490b931f2c6e5e347b9d5b6559ae/src/sentry/grouping/enhancer/__init__.py#L364-L365